scipy/stats/_stats_py.py

```diff
@@ -465,15 +465,18 @@ def _mode_result(mode, count):
     # When a slice is empty, `_axis_nan_policy` automatically produces
     # NaN for `mode` and `count`. This is a reasonable convention for `mode`,
     # but `count` should not be NaN; it should be zero.
-    i = np.isnan(count)
+    xp = array_namespace(mode, count)
+    i = xp.isnan(count)
     if i.shape == ():
-        count = np.asarray(0, dtype=count.dtype)[()] if i else count
+        count = xp.asarray(0, dtype=count.dtype)[()] if i else count
     else:
-        count[i] = 0
+        count = xpx.at(count)[i].set(0)
     return ModeResult(mode, count)
 
 
-@xp_capabilities(np_only=True)
+@xp_capabilities(skip_backends=[('dask.array', "can't compute chunk size"),
+                                ('jax.numpy', "relies on _axis_nan_policy"),
```

Comment:
Previously, I had `jax_jit=False`, so I expected the arrays to be eager. However, it seems they were not, because it was triggering this code:

https://github.com/scipy/scipy/blob/434fc553a403eea11357bfeb0febb66b5175271e/scipy/stats/_axis_nan_policy.py#L439-L444

(The way that `temp` is determined in general is not ideal, since it basically selects only the first (zeroth) argument of the function to check for laziness. But that is exactly right for `mode`, so I don't think that's the problem.)

Why is the array lazy? JAX arrays should be fine when they're eager - they were working back in the original incarnation of gh-22857 before we made them lazy by default.